src/modules/aws_opensearch/entity/domain.ts

```diff
@@ -0,0 +1,134 @@
+import { Max, Min } from 'class-validator';
+import {
+  Check,
+  Column,
+  Entity,
+  JoinColumn,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+
+import {
+  EBSOptions,
+  OpenSearchPartitionInstanceType,
+  OpenSearchWarmPartitionInstanceType,
+} from '@aws-sdk/client-opensearch';
+
+import { Policy } from '../../../services/canonical-iam-policy';
+import { cloudId } from '../../../services/cloud-id';
+import { AwsRegions } from '../../aws_account/entity';
+import { Certificate } from '../../aws_acm/entity';
+import { SecurityGroup } from '../../aws_security_group/entity';
+import { Subnet } from '../../aws_vpc/entity';
+
+@Entity()
+export class Domain {
+  /**
+   * @private
+   * Auto-incremented ID field
+   */
+  @PrimaryGeneratedColumn()
+  id?: number;
+
+  /**
+   * @public
+   * Name to identify the domain
+   */
+  @Column({ nullable: false })
+  @cloudId
+  domainName: string;
+
+  @Column({ nullable: true })
```

Comment:
you are missing the description here on several fields, it means it will not be rendered properly on the documentation